.travis.yml

```diff
@@ -25,8 +25,44 @@ jobs:
         - tests/scripts/all.sh -k build_arm_linux_gnueabi_gcc_arm5vte build_arm_none_eabi_gcc_m0plus
 
     - name: full configuration
+      os: linux
+      dist: focal
+      addons:
+        apt:
+          packages:
+          - clang-10
+          - gnutls-bin
       script:
-        - tests/scripts/all.sh -k test_full_cmake_gcc_asan
+        # Do a manual build+test sequence rather than using all.sh,
+        # because there's no all.sh component that does what we want,
+        # which is a build with Clang >= 10 and ASan, running all the SSL
+        # testing.
+        #   - The clang executable in the default PATH is Clang 7 on
+        #     Travis's focal instances, but we want Clang >= 10.
+        #   - Running all the SSL testing requires a specific set of
+        #     OpenSSL and GnuTLS versions and we don't want to bother
+        #     with those on Travis.
+        # So we explicitly select clang-10 as the compiler, and we
+        # have ad hoc restrictions on SSL testing based on what is
+        # passing at the time of writing. We will remove these limitations
+        # gradually.
+        - make generated_files
+        - make CC=clang-10 CFLAGS='-Werror -Wall -Wextra -fsanitize=address,undefined -fno-sanitize-recover=all -O2' LDFLAGS='-Werror -Wall -Wextra -fsanitize=address,undefined -fno-sanitize-recover=all'
+        - make test
+        - programs/test/selftest
+        - tests/scripts/test_psa_constant_names.py
+        # Exclude some tests that use SHA-1 certificates which modern OpenSSL
+        # rejects.
+        - tests/ssl-opt.sh -e 'keyUsage cli.*RSA'
+        # Modern OpenSSL does not support fixed ECDH or null ciphers.
+        # ARIA-128-GCM should work (and ARIA-256-GCM does) but m->O
+        # fails with "dh key too small" only for ARIA-128-GCM!
+        - tests/compat.sh -p OpenSSL -e 'NULL\|ECDH-\|ARIA-128-GCM'
```

Comment:
I'm not running `compat.sh -p mbedTLS` because it doesn't find much that unit tests wouldn't find. Running the interop tests is useful because it's testing interop with the focal versions of openssl and gnutls.